src/csharp/Microsoft.Spark.Extensions.Delta.UnitTest/DeltaTableTests.cs

```diff
@@ -0,0 +1,180 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.Spark.E2ETest;
+using Microsoft.Spark.E2ETest.Utils;
+using Microsoft.Spark.Extensions.Delta.Tables;
+using Microsoft.Spark.Sql;
+using Xunit;
+
+namespace Microsoft.Spark.Extensions.Delta.UnitTest
+{
+    /// <summary>
+    /// These tests will fail if the Delta Jar is not installed in the 
+    /// $SPARK_HOME/jars/ directory.
+    /// </summary>
+    [Collection(Constants.DeltaTestContainerName)]
+    public class DeltaTableTests
+    {
+        private readonly SparkSession _spark;
+
+        public DeltaTableTests(SparkFixture fixture)
+        {
+            _spark = fixture.Spark;
+        }
+
+        /// <summary>
+        /// Run the end-to-end scenario from the Delta Quickstart tutorial.
+        /// </summary>
+        /// <see cref="https://docs.delta.io/latest/quick-start.html"/>
+        [Fact]
+        public void TestTutorialScenario()
+        {
+            using TemporaryDirectory tempDirectory = new TemporaryDirectory();
+            string path = Path.Combine(tempDirectory.Path, "delta-table");
+
+            // Write data to a Delta table.
+            DataFrame data = _spark.Range(0, 5);
+            data.Write().Format("delta").Save(path);
+
+            // Validate that data contains the the sequence [0 ... 4].
+            ValidateTutorialDataFrame(Enumerable.Range(0, 5), data);
+
+            // Create a second iteration of the table.
+            data = _spark.Range(5, 10);
+            data.Write().Format("delta").Mode("overwrite").Save(path);
+
+            // Load the data into a DeltaTable object.
+            var deltaTable = DeltaTable.ForPath(path);
+
+            // Validate that deltaTable contains the the sequence [5 ... 9].
+            ValidateTutorialDataFrame(Enumerable.Range(5, 5), deltaTable.ToDF());
```

Comment:
If this is supposed to validate [5 .. 9] why are we passing `Enumerable.Range(5, 5)`

Reply:
The params for `Range()` are starting index and number of elements.